tls/s2n_async_offload.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+#include "tls/s2n_async_offload.h"
+
+#include "api/s2n.h"
+#include "crypto/s2n_hash.h"
+#include "crypto/s2n_signature.h"
+#include "error/s2n_errno.h"
+#include "tls/s2n_connection.h"
+#include "tls/s2n_handshake.h"
+#include "utils/s2n_blob.h"
+#include "utils/s2n_mem.h"
+#include "utils/s2n_result.h"
+#include "utils/s2n_safety.h"
+
+S2N_RESULT s2n_async_offload_cb_invoke(struct s2n_connection *conn, struct s2n_async_op *op)
```

Comment:
I wonder if it'd be worth doing some validation to make sure the caller set up their `s2n_async_op` properly? Like we wouldn't want to invoke the user's callback without a perform callback set, for example.